### ISSUE TITLE
StateVersionOutput fails to unmarshal

### ISSUE THREAD

**Issue by thegrumpylion** [hashicorp/go-tfe#231]
The tests

```go
package controllers

import (
        "context"
        "fmt"
        "testing"

        "github.com/hashicorp/go-tfe"
)

func getClient() (*tfe.Client, error) {
        cfg := &tfe.Config{
                Token: token,
        }
        cli, err := tfe.NewClient(cfg)
        if err != nil {
                return nil, err
        }
        return cli, nil
}

func TestCurrent(t *testing.T) {

        cli, err := getClient()
        if err != nil {
                t.Fatal(err)
        }
        ctx := context.Background()

        ws, err := cli.Workspaces.Read(ctx, "myorg", "myworkspace")
        if err != nil {
                t.Fatal(err)
        }

        st, err := cli.StateVersions.Current(ctx, ws.ID)
        if err != nil {
                t.Fatal(err)
        }

        for _, o := range st.Outputs {
                fmt.Println(o)
        }
}

func TestCurrentOutputs(t *testing.T) {

        cli, err := getClient()
        if err != nil {
                t.Fatal(err)
        }

        ctx := context.Background()

        ws, err := cli.Workspaces.Read(ctx, "myorg", "myworkspace")
        if err != nil {
                t.Fatal(err)
        }

        st, err := cli.StateVersions.Current(ctx, ws.ID)
        if err != nil {
                t.Fatal(err)
        }

        for _, o := range st.Outputs {
                fmt.Println(o)
                out, err := cli.StateVersionOutputs.Read(ctx, o.ID)
                if err != nil {
                        t.Fatal(err)
                }
                fmt.Println(out)
        }
}

func TestCurrentWithOptions(t *testing.T) {

        cli, err := getClient()
        if err != nil {
                t.Fatal(err)
        }

        ctx := context.Background()

        ws, err := cli.Workspaces.Read(ctx, "myorg", "myworkspace")
        if err != nil {
                t.Fatal(err)
        }

        st, err := cli.StateVersions.CurrentWithOptions(ctx, ws.ID, &tfe.StateVersionCurrentOptions{
                Include: "outputs",
        })
        if err != nil {
                t.Fatal(err)
        }

        for _, o := range st.Outputs {
                fmt.Println(o)
        }
}
```

The output

```sh
=== RUN   TestCurrent
&{wsout-aaaa  false  }
&{wsout-bbbb false  }
--- PASS: TestCurrent (1.25s)
=== RUN   TestCurrentOutputs
&{wsout-aaaa false  }
    stateversionoutputs_test.go:69: Invalid type provided
--- FAIL: TestCurrentOutputs (1.47s)
=== RUN   TestCurrentWithOptions
    stateversionoutputs_test.go:94: Invalid type provided
--- FAIL: TestCurrentWithOptions (1.21s)
FAIL
exit status 1
```

After some debugging, I noticed that the server returns the response but unmarshalling fails.

I also tried to `curl` the 2 outputs with the same token and endpoint generated by `go-tfe` and it works.

```sh
curl --header "Authorization: Bearer $TOKEN" https://app.terraform.io/api/v2/state-version-outputs/wsout-aaaa
```

```
{
  "data": {
    "id": "wsout-aaaa",
    "type": "state-version-outputs",
    "attributes": {
      "name": "the_secret",
      "sensitive": true,
      "type": "object",
      "value": {
        ...
      }
    },
    "links": {
      "self": "/api/v2/state-version-outputs/wsout-aaaa"
    }
  }
}
```

**Comment by thegrumpylion** [comment]
* os: Ubuntu 20.04.2
* lib version: github.com/hashicorp/go-tfe v0.15.0
* go version: go1.16 linux/amd64

**Comment by thegrumpylion** [comment]
Ok looks like it breaks for `"type": "object"`, if I use only string outputs it works.

**Comment by Houndie** [comment]
It looks like there was a PR in January that proposes to fix this, although like this Issue, the silence coming from the library maintainers is deafening.

**Comment by omarismail** [comment]
Hey @thegrumpylion, we know this has been a long standing issue and apologies for the delay. We have addressed this problem in #246. Now the type of `StateVersionOutput.Value` is an `interface{}` and you can type cast it into anything you'd like.
